condor.go

```diff
@@ -157,63 +149,25 @@ func (cl *CondorLauncher) launch(s *model.Job, condorPath, condorConfig string)
 	if err != nil {
 		return "", errors.Wrapf(err, "failed to create the directory %s", sdir)
 	}
+
+	// Store the Condor configs in Vault.
 	childToken, err := cl.storeConfig(s)
 	if err != nil {
 		return "", err
 	}
+
+	// Create a copy of the configuration that also contains the Vault child token.
 	cfgCopy := CopyConfig(cl.cfg)
 	cfgCopy.Set("vault.child_token.token", childToken)
-	subfiles := []struct {
-		filename    string
-		filecontent []byte
-		template    *template.Template
-		data        interface{}
-		skipTmpl    bool // skip applying the data field to the template
-		jsonify     bool // marshal the data field as JSON and store it in the filecontent field
-		permissions os.FileMode
-	}{
-		{
-			filename:    path.Join(sdir, "iplant.cmd"),
-			template:    SubmissionTemplate,
-			data:        s,
-			permissions: 0644,
-		},
-		{
-			filename:    path.Join(sdir, "config"),
-			template:    JobConfigTemplate,
-			data:        cfgCopy,
-			permissions: 0644,
-		},
-		{
-			filename:    path.Join(sdir, "job"),
-			data:        s,
-			skipTmpl:    true,
-			jsonify:     true,
-			permissions: 0644,
-		},
-	}
 
-	for _, sf := range subfiles {
-		var fileContent *bytes.Buffer
-		if !sf.skipTmpl {
-			fileContent, err = GenerateFile(sf.template, sf.data)
-			if err != nil {
-				return "", err
-			}
-			sf.filecontent = fileContent.Bytes()
-		}
-		if sf.jsonify {
-			sf.filecontent, err = json.Marshal(sf.data)
-			if err != nil {
-				return "", nil
-			}
-		}
-		err = ioutil.WriteFile(sf.filename, sf.filecontent, sf.permissions)
-		if err != nil {
-			return "", errors.Wrapf(err, "failed to write to file %s", sf.filename)
-		}
+	// Generate the submission files, always using the condor job submission format for now.
+	jobSubmissionBuilder, err := jobs.NewJobSubmissionBuilder("condor", cfgCopy)
+	if err != nil {
+		return "", err
 	}
-	submissionPath := subfiles[0].filename
+	submissionPath, err := jobSubmissionBuilder.Build(s, sdir)
```

Comment:
Should any potential returned `err` be checked here?

Reply:
Ah, yes it should. Good catch!